lib/events.js

```diff
@@ -84,8 +107,35 @@ EventEmitter.init = function() {
   }
 
   this._maxListeners = this._maxListeners || undefined;
+
+  if (opts && opts.autoCatch) {
+    this[kApply] = applyAndCatch;
+  }
 };
 
+const apply = Reflect.apply;
+function applyAndCatch(handler, that, args) {
+  const promise = apply(handler, that, args);
+  if (promise && typeof promise.then === 'function') {
+    promise.then(undefined, (err) => {
+      // The callback is called with nextTick to avoid a follow-up
```

Comment:
I'd use different terminology than "avoid a follow up action" here.

Reply:
Would you mind making a suggestion?

Reply:
I would if I had one, sorry :]

Reply:
I've updated the text. Let me know what you think!